xarray/tests/test_groupby.py

```diff
@@ -1245,12 +1245,19 @@ def test_groupby_iter(self):
             ):
                 assert_identical(exp_dv, act_dv)
 
-    def test_groupby_properties(self):
+    def test_groupby_properties(self) -> None:
         grouped = self.da.groupby("abc")
         expected_groups = {"a": range(0, 9), "c": [9], "b": range(10, 20)}
         assert expected_groups.keys() == grouped.groups.keys()
         for key in expected_groups:
-            assert_array_equal(expected_groups[key], grouped.groups[key])
+            expected_group = expected_groups[key]
+            actual_group = grouped.groups[key]
+
+            # TODO: array_api doesn't allow slice:
+            assert not isinstance(expected_group, slice)
+            assert not isinstance(actual_group, slice)
+
+            np.testing.assert_array_equal(expected_group, actual_group)
```

Comment:
Not sure why we can get slices from `da.groupby("abc").groups` but slices are not considered an ArrayLike and fails using `np.array_api.asarray`.

Reply:
Using slices when possible is an optimization. This particularl groupby call doesn't have any so your changes are :+1:

Reply:
Cool, if it's useful it's fine. Shouldn't expect `xp.asarray` to work though.